mcs/class/corlib/Test/System.Threading/WaitHandleTest.cs

```diff
@@ -395,6 +395,30 @@ public void InterrupedWaitOne ()
 			}
 		}
 
+		[Test]
+		public void InterruptedWaitOneWithTimeout ()
+		{
+			using (ManualResetEvent mre = new ManualResetEvent (false))
+			using (ManualResetEvent ready = new ManualResetEvent (false)) {
+				var thread = new Thread (() => {
+					ready.Set ();
+					mre.WaitOne (1000);
+				});
+
+				thread.Start ();
+				ready.WaitOne ();
+
+				Thread.Sleep (10); // wait a bit so we enter mre.WaitOne
+
+				DateTime start = DateTime.Now;
+				while (DateTime.Now < start.AddMilliseconds (500)) {
```

Comment:
nit: I think it'd be a bit clearer this way:

```csharp
DateTime end = DateTime.Now.AddMilliseconds (500);
while (DateTime.Now < end) {
...
```

... and it also doesn't trigger my inner OCD against adding the 500ms millions of times in the loop (yes I know it's totally irelevant here :laughing:)